insights/core/fava.py

```diff
@@ -614,3 +617,150 @@ def create_module(full_module_name, the_ast, the_globals):
         package.__dict__[module_element_name] = mod
 
     return mod
+
+
+class FavaImporter(object):
+    """ Hook into python's import machinery so standard import statements
+        can be used for .fava files.
+    """
+    ext = ".fava"
+
+    def __init__(self, path):
+        path = os.path.realpath(path)
+        if os.path.exists(path):
+            # punt if we don't see any fava files in the path
+            if not any(n.endswith(self.ext) for n in os.listdir(path)):
+                raise ImportError()
+            self.path = path
+
+    def find_module(self, fullname, paths=None):
+        """ Returns a FavaLoader for fava files and a pkgutil.ImpLoader
+            for standard python files. The ImpLoader is a PEP-302 wrapper
+            for python's regular import machinery.
+        """
+        name = fullname.split(".")[-1]
+        filename = os.path.join(self.path, name + self.ext)
+        if os.path.exists(filename):
+            return FavaLoader(filename)
+        return pkgutil.ImpLoader(fullname, *imp.find_module(name, [self.path]))
```

Comment:
Do you need to return a custom loader here?  Based on PEP 302, you can let the original import machinery handle the import if you raise `ImportError` here.

Reply:
Raising `ImportError` from `find_module` would abort the import entirely, which isn't what we want: "If find_module() raises an exception, it will be propagated to the caller, aborting the import." When `FavaImporter` successfully constructs, that tells the import machinery that we can handle the path. Since I wanted to support .fava and .py files in the same package, find_module has to return the right loader for either of them.

Reply:
I see, thanks for clarifying.